cmd/siac/walletcmd.go

```diff
@@ -171,19 +171,37 @@ use it instead of displaying the typical interactive prompt.`,
 	}
 )
 
-const askPasswordText = "We need to encrypt the new data using the current wallet password, please provide: "
-
-const currentPasswordText = "Current Password: "
-const newPasswordText = "New Password: "
+const askPasswordText = "Need to encrypt new data with wallet password"
 
 // passwordPrompt securely reads a password from stdin.
 func passwordPrompt(prompt string) (string, error) {
-	fmt.Print(prompt)
+	fmt.Print(prompt, ": ")
 	pw, err := terminal.ReadPassword(0)
 	fmt.Println()
 	return string(pw), err
 }
 
+// same except handle error here
+func passPrompt(prompt string) string {
+	pass, err := passwordPrompt(prompt)
+	if err != nil {
+		die("Reading", prompt, "failed:", err)
+	}
+	return pass
+}
+
+// ask & confirm a new password with min length
+func passConfirm(prompt string) string {
+	pass := passPrompt(prompt)
+	pass2 := passPrompt(prompt + " (again)")
+
+	if len(pass) < 8 || pass != pass2 {
+		die(prompt, "mismatch or too short")
```

Comment:
for better ux, this could be two separate if-statements so that the error message is clear

`len(pass) < 8` "too short"
`pass != pass2` "mismatch"